### ISSUE TITLE
Consider building Python with -mcpu optimization

### ISSUE THREAD

**Issue by ThinkOpenly** [advancetoolchain/advance-toolchain#2233]
Python does not have a multilib capability, but one could theoretically be imposed on it.  The bulk of the Python runtime is in a common `lilbpython.so` file.  It should be possible to build different versions of this file with different compiler options and push them into cpu-specific (HWCAP-specific) directories, like:
```
/lib64/libpython.so
/lib64/power9/libpython.so
/lib64/power10/libpython.so
```
...and the optimal one for the current platform should be chosen automatically by the loader.

**Comment by ThinkOpenly** [comment]
I'll note here that while it seems there is a fairly easy method to provide architecture-optimized `libpython.so` files, there are also Python modules for which a mechanism is not straightforward.  From what I see, these are "always" found in `{exec_prefix}/{platlibdir}/python{major.minor}/lib-dynload`. The system loader is not invoked, so there are no HWCAP-like alternative directory searches involved, like for `.../lib64/power9`. This logic is all embedded within the `libpython.so`, so perhaps an architecture-specific `libpython.so` could be configured with an alternate directory.  There is a lot of platform- and architecture-independent code above `lib-dynload`, so changing just that to an alternative name seems feasible.  Unfortunately `lib-dynload` is hard-coded in a handful of places, so an out-of-tree patch to change that or allow a command-line override would be needed.

**Comment by ThinkOpenly** [comment]
I started looking into this earnestly. Unfortunately, running in a mixed environment, with a `power8` (non-pcrel) `python` binary using a `power10` (pcrel) `libpython.so` calling a `power8` (non-pcrel) Python module hits an apparent GCC issue: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=102107. So, the "easy" part of this work is stalled for a bit.

**Comment by tuliom** [comment]
Can we close this now after PR #2351 has been merged?

**Comment by ThinkOpenly** [comment]
> Can we close this now after PR #2351 has been merged?

Indeed. Fixed by PR #2351 .
